src/catimg.c

```diff
@@ -111,18 +140,31 @@ int main(int argc, char *argv[])
             precision = 1;
     }
 
-    if (cols < 1) // if precision is 2 we can use the terminal full width. Otherwise we can only use half
-        cols = terminal_columns() / (2 / precision);
+    // if precision is 2 we can use the terminal full width/height. Otherwise we can only use half
+    max_cols = terminal_columns() / (2 / precision);
+    max_rows = terminal_rows() * 2 / (2 / precision);
 
     if (strcmp(file, "-") == 0) {
         img_load_from_stdin(&img);
     } else {
         img_load_from_file(&img, file);
     }
-    if (cols < img.width) {
-        float sc = cols/(float)img.width;
+    if (cols == 0 && rows == 0) {
+        sc = max_cols/(float)img.width;
```

Comment:
can you put a span around mathematical operators like in the rest of the codebase?

Reply:
What does "a span" mean? Like spaces before and after '/'?